astropy/conftest.py

```diff
@@ -52,6 +65,9 @@ def pytest_unconfigure(config):
         matplotlib.rcParams.update(matplotlibrc_cache)
         matplotlibrc_cache.clear()
 
+    os.environ.pop('XDG_CONFIG_HOME')
+    os.environ.pop('XDG_CACHE_HOME')
+
```

Comment:
Why is this needed here but not in the other conftest files ?

Reply:
I don't know how I feel about playing with env var in general in `conftest`. Let's say user sets custom vars, then run tests, `conftest` messes them up, now user complains that the these vars are not what they set anymore. Can this happen?

Reply:
The reason it's needed here and not in the other ``conftest.py`` files is because this is the only one which might be used inside a persistent python session such as IPython. The other ones will only matter if you run ``pytest``, and the Python process will end at the end of the tests. Any changes to ``os.environ`` don't change the variables beyond the Python process:

```
Python 3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 03:13:28) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.environ['FOO'] = 'BAR'
>>> 
$ echo $FOO
$
```

However, in this specific ``conftest.py`` file I should probably indeed save any pre-existing variables and reset then in ``unconfigure``.